src/contract_error.ts

```diff
@@ -0,0 +1,77 @@
+import { xdr } from 'stellar-sdk';
+
+export class ContractError extends Error {
+  public type?: string;
+  constructor(type: string, message: string) {
+    super(message);
+    this.type = type;
+  }
+}
+
+export enum BlendErrors {
```

Comment:
If we return the type as an enum, it likely makes sense to rename this as something more generic.

`ContractErrorType` maybe